src/lvl_script_commands.c

```diff
@@ -0,0 +1,2108 @@
+  }
+}
+
+static void command_kill_creature(long plr_range_id, const char *crtr_name, const char *criteria, int count)
+{
+    SCRIPTDBG(11, "Starting");
+    if (count <= 0)
+    {
+        SCRPTERRLOG("Bad creatures count, %d", count);
+        return;
+  }
+  long crtr_id = parse_creature_name(crtr_name);
+  if (crtr_id == CREATURE_NONE) {
+    SCRPTERRLOG("Unknown creature, '%s'", crtr_name);
+    return;
+  }
+  long select_id = parse_criteria(criteria);
+  if (select_id == -1)
+  {
+    SCRPTERRLOG("Unknown select criteria, '%s'", criteria);
+    return;
+  }
+  command_add_value(Cmd_KILL_CREATURE, plr_range_id, crtr_id, select_id, count);
+}
+
+static void command_level_up_creature(long plr_range_id, const char *crtr_name, const char *criteria, int count)
+{
+    SCRIPTDBG(11, "Starting");
+    if (count <= 0)
+    {
+        SCRPTERRLOG("Bad count, %d", count);
+        return;
+  }
+  long crtr_id = parse_creature_name(crtr_name);
+  if (crtr_id == CREATURE_NONE)
+  {
+    SCRPTERRLOG("Unknown creature, '%s'", crtr_name);
+    return;
+  }
+  long select_id = parse_criteria(criteria);
+  if (select_id == -1) {
+    SCRPTERRLOG("Unknown select criteria, '%s'", criteria);
+    return;
+  }
+  if (count < 1)
+  {
+    SCRPTERRLOG("Parameter has no positive value; discarding command");
+    return;
+  }
+  if (count > 9)
+  {
+      count = 9;
+  }
+  command_add_value(Cmd_LEVEL_UP_CREATURE, plr_range_id, crtr_id, select_id, count);
+}
+
+static void command_use_power_on_creature(long plr_range_id, const char *crtr_name, const char *criteria, long caster_plyr_idx, const char *magname, int splevel, char free)
+{
+  SCRIPTDBG(11, "Starting");
+  if (splevel < 1)
+  {
+    SCRPTWRNLOG("Spell %s level too low: %d, setting to 1.", magname, splevel);
+    splevel = 1;
+  }
+  if (splevel > MAGIC_OVERCHARGE_LEVELS)
+  {
+    SCRPTWRNLOG("Spell %s level too high: %d, setting to %d.", magname, splevel, MAGIC_OVERCHARGE_LEVELS);
+    splevel = MAGIC_OVERCHARGE_LEVELS;
+  }
+  splevel--;
+  long mag_id = get_rid(power_desc, magname);
+  if (mag_id == -1)
+  {
+    SCRPTERRLOG("Unknown magic, '%s'", magname);
+    return;
+  }
+  long crtr_id = parse_creature_name(crtr_name);
+  if (crtr_id == CREATURE_NONE) {
+    SCRPTERRLOG("Unknown creature, '%s'", crtr_name);
+    return;
+  }
+  long select_id = parse_criteria(criteria);
+  if (select_id == -1) {
+    SCRPTERRLOG("Unknown select criteria, '%s'", criteria);
+    return;
+  }
+  PowerKind pwr = mag_id;
+  if((PlayerNumber) caster_plyr_idx > PLAYER3)
+  {
+    if(pwr == PwrK_CALL2ARMS || pwr == PwrK_LIGHTNING)
+    {
+        SCRPTERRLOG("Only players 0-3 can cast %s", magname);
+        return;
+    }
+  }
+
+  // encode params: free, magic, caster, level -> into 4xbyte: FMCL
+  long fmcl_bytes;
+  {
+      signed char f = free, m = mag_id, c = caster_plyr_idx, lvl = splevel;
+      fmcl_bytes = (f << 24) | (m << 16) | (c << 8) | lvl;
```

Comment:
Consider use new syntax (#_check + #_process) and use `value->bytes[...]`

Reply:
I'm just focussing on moving fuctions to better locations then looking at the actual content of said logic, maybe that kind of stuff could be in a seperate refactor

Reply:
ok